src/katgpucbf/xbgpu/main.py

```diff
@@ -123,14 +123,23 @@ def parse_beam(value: str) -> BOutput:
 
     - name
     - dst
+    - pol
     """
 
     def _field_callback(kws: BOutputDict, key: str, data: str) -> None:
-        raise ValueError(f"unknown key {key}")
+        match key:
+            case "pol":
+                kws[key] = int(data)
+                if kws[key] not in {0, 1}:
+                    raise ValueError("pol must be either 0 or 1")
```

Comment:
Worth asking, would it be equally acceptable to do `if kws[key] not in {0, 1}` *before* making the assignment?
* i.e. Why check after we've stored it?

Reply:
Sounds like you're suggesting
```python
                if kws[key] not in {0, 1}:
                    raise ValueError("pol must be either 0 or 1")
                kws[key] = int(data)
```
That won't work because `kws[key]` won't exist yet.

I guess one could do
```python
                value = int(data)
                if value not in {0, 1}:
                    raise ValueError("pol must be either 0 or 1")
                kws[key] = value
```
but that's more code.

Reply:
Argh, too right. Or even
```python
if int(data) not in {0, 1}:
    raise ValueError("pol must be either 0 or 1")
kws[key] = int(data)
```

But I gather there is some unease around using a naked `int(data)`.